cpp/katz_centrality_module/katz_centrality_online_module.cpp

```diff
@@ -59,59 +57,91 @@ void SetKatzCentrality(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *re
     auto katz_centralities = katz_alg::SetKatz(*graph, alpha, epsilon);
 
     for (auto &[vertex_id, centrality] : katz_centralities) {
-      // Insert the Katz centrality record
       InsertKatzRecord(memgraph_graph, result, memory, centrality, vertex_id);
     }
   } catch (const std::exception &e) {
-    // We must not let any exceptions out of our module.
     mgp::result_set_error_msg(result, e.what());
     return;
   }
 }
 
 void UpdateKatzCentrality(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
   try {
-    // Created vertices
-    auto created_vertices = mg_utility::GetNodeIDs(mgp::value_get_list(mgp::list_at(args, 0)));
-    auto created_edges = mg_utility::GetEdgeEndpointIDs(mgp::value_get_list(mgp::list_at(args, 1)));
-    auto created_edge_ids = mg_utility::GetEdgeIDs(mgp::value_get_list(mgp::list_at(args, 1)));
-
-    // Deleted entities
-    auto deleted_vertices = mg_utility::GetNodeIDs(mgp::value_get_list(mgp::list_at(args, 2)));
-    auto deleted_edges = mg_utility::GetEdgeEndpointIDs(mgp::value_get_list(mgp::list_at(args, 3)));
-
-    auto graph = mg_utility::GetGraphView(memgraph_graph, result, memory, mg_graph::GraphType::kDirectedGraph);
-    std::transform(created_edge_ids.begin(), created_edge_ids.end(), created_edge_ids.begin(),
-                   [&graph](std::uint64_t id) -> std::uint64_t { return graph.get()->GetInnerEdgeId(id); });
-
-    auto katz_centralities = katz_alg::UpdateKatz(*graph, created_vertices, created_edges, created_edge_ids,
-                                                  deleted_vertices, deleted_edges);
-
-    for (auto &[vertex_id, centrality] : katz_centralities) {
-      // Insert the Katz centrality record
-      InsertKatzRecord(memgraph_graph, result, memory, centrality, vertex_id);
+    mgp::memory = memory;
+
+    const auto record_factory = mgp::RecordFactory(result);
+    const auto graph = mgp::Graph(memgraph_graph);
+    const auto arguments = mgp::List(args);
+
+    std::vector<std::pair<uint64_t, double>> katz_centralities;
+
+    if (katz_alg::NoPreviousData()) {
+      auto legacy_graph = mg_utility::GetGraphView(memgraph_graph, result, memory, mg_graph::GraphType::kDirectedGraph);
+      katz_centralities = katz_alg::SetKatz(*legacy_graph);
+
+      for (auto &[node_id, centrality] : katz_centralities) {
+        auto record = record_factory.NewRecord();
+        record.Insert(kFieldNode, graph.GetNodeById(mgp::Id::FromUint(node_id)));
+        record.Insert(kFieldRank, centrality);
+      }
```

Comment:
you can restructure this part to like this:
if (katz_alg::NoPreviousData()) {
      ...
    return
}
<no else statement here>